app/src/main/java/com/example/pricingpal/model/CSVReader.kt

```diff
@@ -0,0 +1,80 @@
+package com.example.pricingpal.model
+
+import android.content.Context
+import java.io.FileNotFoundException
+import java.io.IOException
+import java.io.InputStream
+import java.io.InputStreamReader
+import java.util.Scanner
+
+/**
+ * Class: CSVReader
+ * @author Julian Ellis
+ * @version 2.1
+ * @written 9/24/2023
+ * This class extends AppCompatActivity and holds a function to read a CSV file
+ *   from an assets folder.
+ *
+ */
+class CSVReader {
+
+    /**
+     * readFile
+     * This function reads a csv file into an ArrayList of Item objects. This
+     *  is a composable function.
+     * @param fileName The name of the file to be read
+     * @param c Context of the calling program (In MainActivity.kt use "this" keyword)
+     * @return itemList The list of Item objects generated by the function
+     */
+
+    fun readFile(c: Context, fileName: String): List<Item> {
+
+        var itemList = ArrayList<Item>()
+
+        try {
+            //Point the input stream at the assets folder and open the file
+            val inputStream: InputStream = c.assets.open(fileName)
+            val n = Scanner(InputStreamReader(inputStream))
+            itemList = ArrayList<Item>()
+            //Skip the first line if column headers are present, otherwise comment out
+            //NOTE: testdata.csv has no column headers
+            //val skipColumnHeader = n.nextLine()
+            //Read file and put into Item objects list
+            while (n.hasNext()) {
+                val line = n.nextLine()
+                val row = line.split(",")
+                //This error should resolve itself when updated item class is added
+                val item = Item(row[0], row[1], row[2].toDouble())
+                itemList.add(item)
+            }
+            n.close()
+        } catch (e: FileNotFoundException) {
+            println("The file was not found. Check file name or path")
+        } catch (e: IOException) {
+            println("There was an error reading the file")
+        } catch (e: NullPointerException) {
+            println("There is nothing in the list")
+        }
+        //Return the list of items generated in the function
+        return itemList
+    }
+
+
+    //Original algorithm
```

Comment:
Does this need to be here? Removing obsolete code makes things neater.

Reply:
I'd like to leave this here for reference for now. It's just the simplified algorithm to read the file without all the try catch and other stuff. It makes it easier to see what it does. It can be deleted at some point

Reply:
Since it is an exact duplicate of the code above it, my position is that it should probably not be added to the main branch. The code in use is perfectly legible, especially with your comments, so the comment doesn't really serve a purpose. But if you feel strongly that it adds value, it can be included.

I'll leave this conversation as unresolved for a bit, just in case we want to discuss further or anyone else wants to weigh in.

Reply:
For now, for ease of understanding for everyone, if we could leave it as to be deleted later I think it has value for that purpose. If everyone agrees to go ahead and remove it I will do so

Reply:
Ok I took it out. I have it locally if it's needed for reference